cscs-checks/analytics/hpcg/hpcg_benchmark.py

```diff
@@ -0,0 +1,138 @@
+                'perf': (13.4, -0.1, None)
+            },
+        }
+
+        self.perf_patterns = {
+            'perf': sn.extractsingle(
+                r'HPCG result is VALID with a GFLOP\/s rating of=\s*'
+                r'(?P<perf>\S+)', output_file, 'perf',  float)
+        }
+        self.maintainers = ['SK']
+        self.tags = {'diagnostic'}
+
+    def setup(self, partition, environ, **job_opts):
+        self.num_tasks = self.system_num_tasks[self.current_system.name
+                                               + ":" + partition.name]
+        super().setup(partition, environ, **job_opts)
+
+@rfm.simple_test
+class HPCGCheckMKL(rfm.RegressionTest):
+    def __init__(self):
+        super().__init__()
+
+        self.descr = 'HPCG benchmark Intel MKL implementation'
+        self.valid_systems = ['daint:mc', 'dom:mc', 'daint:gpu', 'dom:gpu']
+        self.valid_prog_environs = ['PrgEnv-intel']
+        self.modules = ['craype-hugepages8M']
+        #self.sourcesdir needed for "CrayXC" config file
+        self.build_system = 'Make'
+        self.prebuild_cmd = ['cp -r ${MKLROOT}/benchmarks/hpcg/* .',
+                             'mv Make.CrayXC setup',
+                             './configure CrayXC']
+
+        self.num_tasks = 0
+        self.num_tasks_per_core = 2
+        self.problem_size = 104
+
+        self.variables  = {
+            'HUGETLB_VERBOSE': '0',
+            'MPICH_MAX_THREAD_SAFETY' : 'multiple',
+            'MPICH_USE_DMAPP_COLL': '1',
+            'PMI_NO_FORK': '1',
+            'KMP_HW_SUBSET' : '9c,2t',
+            'KMP_AFFINITY' : 'granularity=fine,compact'
+        }
+
+        self.executable = 'bin/xhpcg_avx2'
+        self.executable_opts = ['--nx=%d' % self.problem_size,
+                                '--ny=%d' % self.problem_size,
+                                '--nz=%d' % self.problem_size, '-t2']
+        self.sanity_patterns = sn.assert_eq(4, sn.count(
+            sn.findall(r'PASSED', self.outfile_lazy)))
+        self.reference = {
+            'dom:mc': {
+                'perf': (22, -0.1, None)
+            },
+            'daint:mc': {
+                'perf': (22, -0.1, None)
+            },
+            'dom:gpu': {
+                'perf': (10.7, -0.1, None)
+            },
+            'daint:gpu': {
+                'perf': (10.7, -0.1, None)
+            },
+        }
+
+        self.maintainers = ['SK']
+        self.tags = {'diagnostic'}
+
+    @property
+    @sn.sanity_function
+    def num_tasks_assigned(self):
+        return self.job.num_tasks
+
+    @property
+    @sn.sanity_function
+    def outfile_lazy(self):
+        pattern = 'n%d-%dp-%dt-*.yaml' % (self.problem_size,
+                                          self.job.num_tasks,
+                                          self.num_cpus_per_task)
+        return sn.getitem(sn.glob(pattern), 0)
+
+    def setup(self, partition, environ, **job_opts):
+        if partition.name == 'gpu':
```

Comment:
I'd rewrite this as follows in order to be more future-proof:

```suggestion
        if partition.fullname in ['daint:gpu', 'dom:gpu']:
```